src/stk/image/volume.h

```diff
@@ -175,6 +183,8 @@ namespace stk
         float3 _spacing; // Size of a voxel
 
         bool _contiguous;
+
+        std::map<std::string, std::any> _metadata;
```

Comment:
Yeah, this should work.

However, the `map` puts a strain on the copying of `Volume`, so I'm thinking about implementing this as some COW-mechanism because I assume most use-cases will only read the metadata, so similar to `VolumeData`, the metadata is actually shared between volumes, until the user writes to it.

Should be quite easy to implement, a `shared_ptr` holding the data, and whenever the user writes we just check if the ref count is greater than 1, in that case we copy the data first.